packages/dynamite/dynamite/lib/src/builder/client.dart

```diff
@@ -497,111 +511,107 @@ final _response = await ${isRootClient ? 'this' : '_rootClient'}.doRequest(
 ''',
           );
 
-          if (operation.responses != null) {
-            if (operation.responses!.length > 1) {
-              throw Exception('Can not work with multiple status codes right now');
+          for (final responseEntry in responses.entries) {
+            final response = responseEntry.key;
+            final statusCodes = responseEntry.value;
+            code.write(
+              'if (${statusCodes.map((final statusCode) => '_response.statusCode == $statusCode').join(' || ')}) {',
+            );
```

Comment:
Can we use a switch here?
Would make lookup in constant time and I think the code would be cleaner.

Reply:
But in the switch case it would just be ` a || b` again, right? We don't have the fall through cases with Dart 3 anymore.
I don't see how that would benefit us (until we support multiple schemas).

Reply:
> We don't have the fall through cases with Dart 3 anymore.

:woman_facepalming:  I forgot that.